File: Practica 3/Python/Pila.py
class Pila:
    def __init__(self, n):
        self.arreglo = [0]*n
        self.top = -1
        self.n = n            
    
    def push(self, e):
        if self.isFull():
            print("La pila está llena")
        else:
            self.top += 1
            self.arreglo[self.top] = e
    
    def pop(self):
        if self.isEmpty():
            print("La pila está vacía")
            return None
        else:
            elemento = self.arreglo[self.top]
            self.top -= 1
            return elemento
    
    def peek(self):
        if self.isEmpty():
            print("La pila está vacía")
            return None
        else:
            return self.arreglo[self.top]
    
    def isEmpty(self):
        return self.top == -1
    
    def isFull(self):
        return self.top == self.n - 1

pila = Pila(5)
pila.push(10)
pila.push(20)
pila.push(30)
print("Elemento en la cima:", pila.peek())
print("Elemento eliminado:", pila.pop())
print("Elemento en la cima después del pop:", pila.peek())
print("¿Está vacía?:", pila.isEmpty())
print("¿Está llena?:", pila.isFull())

File: Practica 3/Java/Pila/Pila.java
package practica3;

public class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        arreglo = new long[n];
        top = -1;
    }

    public void push(long e) {
        if (isFull()) {
            System.out.println("La pila está llena");
        } else {
            top++;
            arreglo[top] = e;
        }
    }

    public long pop() {
        if (isEmpty()) {
            System.out.println("La pila está vacía");
            return -1;
        } else {
            long elemento = arreglo[top];
            top--;
            return elemento;
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("La pila está vacía");
            return -1;
        } else {
            return arreglo[top];
        }
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }
}

